se-baseball-game

InJun2 opened the pull request:
랜덤숫자 중복제거 및 동작 로직 구현

InJun2 committed:
se-baseball-game 구현 중
이후 랜덤 숫자 중복제거 및 입력 리스트와 랜덤 리스트 비교 구현 예정

InJun2 committed:
랜덤 숫자 생성 시 중복제거 및 리스트 비교 메소드 구현

InJun2 committed:
Merge pull request #2 from InJun2/injun/main

se-baseball-game 구현 중

DongGeon0908 commented on src/main/java/baseball/input/ScanNumber.java:
```diff
@@ -0,0 +1,37 @@
+
+import java.util.Scanner;
+
+public class ScanNumber {
```
숫자를 scan하는 용도의 class일까요??
그렇다면, 차라리 Input이라는 class를 사용하는게 더 적합하지 않을까요?

DongGeon0908 commented on src/main/java/baseball/input/ScanNumber.java:
```diff
@@ -0,0 +1,37 @@
+            String str = sc.nextLine();
+            System.out.println("입력한 숫자 : " + str);
+
+            if(validateNum(str))
+                return Integer.parseInt(str);
```
```suggestion
            if(validateNum(str)){
                return Integer.parseInt(str);
                }
```

DongGeon0908 commented on src/main/java/baseball/input/ScanNumber.java:
```diff
@@ -0,0 +1,37 @@
+    private final Scanner sc = new Scanner(System.in);
+    private final ScanNumList sList;
+
+    public ScanNumber(){
+        sList = new ScanNumList(scan());
+    }
```
계속 scan하면 언제 끝나는 걸까요??

DongGeon0908 commented on src/main/java/baseball/input/list/ScanNumList.java:
```diff
@@ -0,0 +1,18 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ScanNumList {
```
클래스에 자료구조 이름을 쓰는 것을 지양해야 합니다!
--> 만약 바꾼다면, ScanNums가 좋을 것 같아요!

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    private void generateRandomNum(){
+        Set<Integer> numberSet = new HashSet<>();
+
+        while (numberSet.size() < 3) {
+            int number = (int)(Math.random() * 9 + 1);
+            numberSet.add(number);
+        }
+
+        rList = new RandomNumList(numberSet);
```
지역변수로 만들었다가, 전역변수에 추가한 이유가 있을까요?
하나의 변수만 있어도 괜찮지 않을까요?

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,30 @@
+import java.util.Set;
+
+public class RandomNumber {
+    private RandomNumList rList;
```
--> randomNumList (자료구조 네이밍 지양!)

InJun2 replied:
수정하겟습니다!

DongGeon0908 commented on src/main/java/baseball/random/list/RandomNumList.java:
```diff
@@ -0,0 +1,19 @@
+
+    public RandomNumList(Set<Integer> numberSet){
+        randomNumList = new ArrayList<>(numberSet);
+        Collections.shuffle(randomNumList);
```
Shuffle method를 쓴 이유는 무엇일까요?

InJun2 replied:
java 에서 hashset은 처음 숫자로( 테이블 인덱스) 정렬이 되네요. 생성 시 LinkedHashSet으로 무작위 숫자를 생성한대로 저장하고 shuffle제거 하겠습니다

DongGeon0908 commented on src/main/java/baseball/run/RunGame.java:
```diff
@@ -0,0 +1,46 @@
+        }while(!compareList(randomNum, scanNum));
+    }
+
+    private boolean compareList(RandomNumber rNum, ScanNumber sNum) {
```
메서드에 자료구조 명칭을 쓰는 것을 지양합시다!

DongGeon0908 commented on src/main/java/baseball/run/RunGame.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    private boolean compareList(RandomNumber rNum, ScanNumber sNum) {
+        List<Integer> randomNumList = rNum.getList().getRandomNumList();
+        List<Integer> scanNumList = sNum.getList().getScanNumList();
+        int Strike = 0;
+        int Ball = 0;
+
+        for(int i=0; i<randomNumList.size();i++){
+            if(randomNumList.get(i).equals(scanNumList.get(i))){
+                Strike++;
+                continue;
+            }
+            if(randomNumList.contains(scanNumList.get(i))){
+                Ball++;
+            }
+        }
+
+        if (Strike == 3){
+            System.out.println("3 Strike ! Game End");
+            return true;
+        }
```
너무 많은 기능이 하나의 메서드에 존재하네요!!
메서드는 하나의 일만 해야 합니다!

DongGeon0908 commented on src/main/java/baseball/run/RunGame.java:
```diff
@@ -0,0 +1,46 @@
+            return true;
+        }
+
+        System.out.println("Strike : " + Strike + ", Ball : " + Ball);
```
비즈니스 로직과 입출력 로직이 같이 존재하네요!!
분리해보시죠!!!

DongGeon0908 reviewed (commented):
고생하셨습니다!!

1. 입출력 로직을 비즈니스 로직과 분리하기
2. 메서드가 하나의 일만 하도록 변경하기
3. 자료구조 명칭을 class 혹은 method에 사용하지 않기!

InJun2 committed:
피드백 내용 적용
- Collection 명 class, method 에서 명시하지 않기
- 로직 분리
- 불 필요 지역 변수 제거
- Set 사용 구현체 변경 HashSet -> LinkedHashSet ( shuffle method 제거 )

InJun2 committed:
피드백 내용 적용
- Collection 명 class, method 에서 명시하지 않기
- 로직 분리
- 불 필요 지역 변수 제거
- Set 사용 구현체 변경 HashSet -> LinkedHashSet ( shuffle method 제거 )

InJun2 committed:
피드백 내용 적용
- Collection 명 class, method 에서 명시하지 않기
- 로직 분리
- 불 필요 지역 변수 제거
- Set 사용 구현체 변경 HashSet -> LinkedHashSet ( shuffle method 제거 )

InJun2 committed:
Merge pull request #6 from InJun2/injun/work

피드백 사항 적용

DongGeon0908 commented on src/main/java/baseball/input/InputNumber.java:
```diff
@@ -0,0 +1,42 @@
+
+import java.util.Scanner;
+
+public class InputNumber {
+    private final Scanner sc = new Scanner(System.in);
+    private final InputNumbers sList;
+
+    public InputNumber(){
+        sList = inputNumber();
+    }
+
+    private InputNumbers inputNumber(){
+        String str;
+        do{
+            Output.print("Input three Number : ");
+            str = sc.nextLine();
+
+            errMsg(validateNum(str));
+        }while(!validateNum(str));
+
+        return new InputNumbers(Integer.parseInt(str));
+    }
+
+    private boolean validateNum(String str) {
+        return str.length() ==3 && str.matches("[+-]?\\d*(\\.\\d+)?");
+    }
+
+    private void errMsg(boolean bool){
+        if(!bool){
+            Output.print("잘못된 입력입니다");
+        }
+    }
+
+    public InputNumbers getList(){
+        return sList;
+    }
+}
```
사용자 입력에 대한 유효성 검사에 대해 고려하신 부분 정말 멋지네요!!
한가지 아쉬운 점이 있다면, 굳이 InputNumbers라는 객체를 반환하는게 맞나??라는 의문이 생기네요!

DongGeon0908 replied:
요구사항에 따르면 사용자로부터 233 이런식으로 입력을 받으면,
바로바로 그것에 대한 볼, 카운트 결과를 알려주는데 그점에 해당 구현 방법이 이상하다고 느껴지네요

DongGeon0908 commented on src/main/java/baseball/input/list/InputNumbers.java:
```diff
@@ -0,0 +1,18 @@
+public class InputNumbers {
+    private final List<Integer> inputNumList = new ArrayList<>();
+
+    public InputNumbers(int a){
+        inputNumList.add(a/100);
+        inputNumList.add(a%100/10);
+        inputNumList.add(a%10);
+    }
```
메서드로 빼서 구현할 수 있을 것 같아요!
생성자는 무엇일까요????

InJun2 replied:
객체를 생성해주기 위해 초기화하기 위한 메소드?로 알고있어요

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.Set;
+
+public class RandomNumber {
+    private final RandomNumbers rList;
```
네이밍에 조금 더 신경써주세요

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,29 @@
+public class RandomNumber {
+    private final RandomNumbers rList;
+
+	public RandomNumber(){
+        rList = generateRandomNum();
+    }
```
다른 사람이 읽기 쉽게 코드를 정렬하는 것도 중요합니다!

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    private RandomNumbers generateRandomNum(){
+        Set<Integer> numberSet = new LinkedHashSet<>();
```
LinkedHashSet을 쓰신 이유에 대해 설명해주세요!

InJun2 replied:
HashSet을 사용하면 내부적으로 HashMap을 기반으로 구현되어 있는데  hashMap은 디폴트 버킷사이즈가 16으로 지정이 되어 있어 hashCode를 16으로 나눈 나머지를 배열의 index로 사용한다고 함.  해당 배열안에 숫자를 넣고 이 배열 순서대로 출력이 되는데 로드 팩터가 버킷을 추가로 확보하는 것이 아닌이상 1-9까지의 해시코드를 16으로 나누면 나머지에 따라 각 배열에 넣고 해당 배열을 순서대로 출력하면 오름차순으로  출력이되는듯 함..  그래서 그냥 랜덤으로 생성되는 순간 하나씩 순서로 저장해서 Set을 이용하기 위해 사용했음!!

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,29 @@
+        return new RandomNumbers(numberSet);
+    }
+
+    public RandomNumbers getList(){
+        return rList;
+    }
```
네이밍을 잘 짓는것도 매우매우 중요합니다!

DongGeon0908 commented on src/main/java/baseball/random/list/RandomNumbers.java:
```diff
@@ -0,0 +1,17 @@
+import java.util.Set;
+
+public class RandomNumbers {
+    private final List<Integer> randomNumList;
```
전에도 변수에 자료구조 명칭을 쓰는 것을 지양해야 된다고 말씀드린 것 같아용

DongGeon0908 commented on src/main/java/baseball/random/list/RandomNumbers.java:
```diff
@@ -0,0 +1,17 @@
+public class RandomNumbers {
+    private final List<Integer> randomNumList;
+
+    public RandomNumbers(Set<Integer> numberSet){
+        randomNumList = new ArrayList<>(numberSet);
+    }
```
ㅜㅜㅜㅜㅜ 자료구조 노우노우

DongGeon0908 commented on src/main/java/baseball/random/list/RandomNumbers.java:
```diff
@@ -0,0 +1,17 @@
+        randomNumList = new ArrayList<>(numberSet);
+    }
+
+    public List<Integer> getRandomNumList() {
```
여기도....

DongGeon0908 commented on src/main/java/baseball/run/CompareNumber.java:
```diff
@@ -0,0 +1,46 @@
+    private int strike;
+    private int ball;
+
+    CompareNumber(RandomNumber randomNum){
+        do{
+            valueInitialize();
+            compareNumber(randomNum, new InputNumber());
+            Output.print(outputResult());
+        }while(strike!=THREE);
+    }
```
접근제한자가 없는 이유에 대해 설명해주세용

InJun2 replied:
같은 패키지내에 있어서 일부로 접근제한자를 지정을 하지 않았는데 하는게 좋을까요?

DongGeon0908 commented on src/main/java/baseball/run/CompareNumber.java:
```diff
@@ -0,0 +1,46 @@
+        }
+    }
+
+    private String outputResult(){
+        return strike == THREE ? "3 Strike !" : "Strike : " + strike + ", Ball : " + ball;
+    }
```
여기서 출력 형태를 만들어주는게 좋을까요?
Strike와 Ball의 Count 정보를 넘겨주고, 출력은 다른 곳에서 할 수 있지 않을까요?

InJun2 replied:
그렇게 변경해 보겠습니다 코드리뷰 감사합니다

DongGeon0908 reviewed (commented):
고생하셨습니다!!

앞으로 조금 더 신경써야 되는 부분에 대해 정리하면,

1. 네이밍에 대해 조금 더 신경쓰기!!
2. 코드 정렬 및 다른 사람들이 읽기 쉬운 코드 짜기!!
3. 역할과 책임에 맞추어 조금 더 나누기!!
4. README에 본인이 생각한 요구사항 작성하기!!

InJun2 reviewed (commented):
코드 리뷰 감사합니다 우선 제가 생각하는 내용으로 다시 적었고 이후 코드를 다시 생각해보고 피드백 적용 하겠습니다

leeheefull commented on src/main/java/baseball/output/Output.java:
```diff
@@ -0,0 +1,7 @@
+    public static void print(String str){
+        System.out.println(str);
+    }
+}
```
static으로 사용하게 되면 선언하지 않고 어느 곳에서나 사용할 수 있기 때문에,
특정 부분에서만 인스턴스화하여 사용할 수 있도록 만드는 것은 어떨까요?

View는 모델에서는 쓰이지 않고 Controller에서 쓰이기 때문입니다.
MVC 패턴에 대해 찾아보시면 좋을 것 같습니다.

leeheefull commented on src/main/java/baseball/run/CompareNumber.java:
```diff
@@ -0,0 +1,46 @@
+    private int strike;
+    private int ball;
+
+    CompareNumber(RandomNumber randomNum){
+        do{
+            valueInitialize();
+            compareNumber(randomNum, new InputNumber());
+            Output.print(outputResult());
+        }while(strike!=THREE);
+    }
```
일반적으로 생성자에는 로직이 들어가지 않습니다.
생성자는 말 그대로 생성에 쓰이고 로직은 행위를 뜻하기 때문에 메서드로 구현하는 것이 맞다고 생각합니다.

leeheefull commented on src/main/java/baseball/run/RunGame.java:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    public void run() {
+        new CompareNumber(randomNum);
+        Output.print("BASEBALL GAME END!");
```
생성자에 로직이 들어가면 안되는 이유 중 하나인데,
생성자로 생성을 하고 메서드로 행위를 나타내주면 좀 더 가독성이 좋아보이기 때문입니다.

leeheefull reviewed (commented):
수고하셨습니다.

MVC에 대해 찾아보시면 좀 더 수월하게 구현하실 수 있을 것 같아요.
또한, 숫자 야구 게임을 생각하면 필요한 객체를 먼저 떠올리시고 구현하시면 더 좋을 것 같다는 생각이 들었습니다.

객체는 상태와 행위를 가진 것이고 이를 유기적으로 구성해야합니다. 여기서 행위는 메서드이고 상태는 볼, 스트라이크 카운트 인 것 같네요.

이 말을 한 이유는 CompareNumber.java와 같은 객채가 메서드 역할을 한다고 생각이 들었기 때문입니다.

감사합니다.

InJun2 committed:
피드백 내용 적용
- 변수내의 자료구조명 제거
- 생성자에 로직 제거
- 숫자입력 유효성검증에서 중복 숫자 입력 시 오류 추가
- Output 클래스 static 제거

InJun2 committed:
Merge pull request #7 from InJun2/injun/work

피드백 내용 적용

InJun2 committed:
피드백 내용 적용
- README 파일 설계 및 요구사항 내용 추가
- Input 유효성 검증 로직에서 먼저 정수인지 확인 진행

InJun2 committed:
피드백 내용 적용
- README 파일 설계 및 요구사항 내용 추가

InJun2 committed:
Merge pull request #8 from InJun2/injun/work

Injun/work

DongGeon0908 commented on src/main/java/baseball/random/list/RandomNumbers.java:
```diff
@@ -0,0 +1,17 @@
+
+    public RandomNumbers(Set<Integer> numbers) {
+        randomNumbers = new ArrayList<>(numbers);
+    }
```
set으로 받아서 list로 저장하는 이유는 무엇 때문일까요?

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.Set;
+
+public class RandomNumber {
+    private RandomNumbers numbers;
```
RandomNumbers가 RandomNumber를 가져야 하는게 아닌가요??

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,29 @@
+
+    public RandomNumbers getNumbers() {
+        return numbers;
+    }
```
RandomNumber와 RandomNumbers 두개의 class를 나눠서 하는 것보다,
차라리 하나의 클래스에 랜덤한 숫자들을 만들어주는게 어떨까요??

해당 부분을 분리하고 싶으셔서 이렇게 구현한 것 같은데, 억지로 나눈 것 같다고 느껴지네요

DongGeon0908 commented on src/main/java/baseball/input/InputNumber.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.Scanner;
+import java.util.Set;
+
+public class InputNumber {
```
여기서 사용자의 입력을 받아서, 검증을 하는 건 좋은데, 더 많은 일들이 input class에서 이뤄지는 것 같아요!
검증된 숫자들을 반환하는 메서드를 만들고, 반환된 숫자들을 number라는 클래스에서 관리하는건 어떨가요?

DongGeon0908 commented on src/main/java/baseball/input/InputNumber.java:
```diff
@@ -0,0 +1,53 @@
+        return deduplicationNum(str);
+    }
+
+    private boolean deduplicationNum(String str){
+        int number = Integer.parseInt(str);
+        Set<Integer> numbers = new HashSet<>(Arrays.asList(number / 100, number % 100 / 10, number % 10));
+
+        return numbers.size() == 3;
```
여기서 숫자로 set으로 만들었으면, 그대로 반환해도 되는 거 아닌가요???

DongGeon0908 replied:
왜 Boolean을 주는걸까요?

DongGeon0908 commented on src/main/java/baseball/input/list/InputNumbers.java:
```diff
@@ -0,0 +1,22 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class InputNumbers {
+    private final List<Integer> inputNumbers = new ArrayList<>();
+
+    public InputNumbers(int number) {
+        setInputNumbers(number);
+    }
+
+    private void setInputNumbers(int number) {
+        inputNumbers.add(number / 100);
+        inputNumbers.add(number % 100 / 10);
+        inputNumbers.add(number % 10);
+    }
+
+    public List<Integer> getInputNumbers() {
+        return inputNumbers;
+    }
+}
```
해당 클래스에서 List<Integer>를 생성자로 받아서 관리해도 되는거 아닌가요?
앞쪽에서 이미 다 만들었는데, 중복된 로직이 또 보이네요!

DongGeon0908 commented on src/main/java/baseball/run/CompareNumber.java:
```diff
@@ -0,0 +1,54 @@
+public class CompareNumber {
+    private static final int THREE = 3;
+
+    private int strike;
+    private int ball;
+
+    public CompareNumber() {
+        initValue();
+    }
```
```suggestion
    private final int strike;
    private final int ball;

    public CompareNumber() {
        this.strike = 0;
        this.ball = 0;
    }
```

DongGeon0908 replied:
initValuer가 필요하나요?

DongGeon0908 commented on src/main/java/baseball/run/CompareNumber.java:
```diff
@@ -0,0 +1,54 @@
+        ball = 0;
+    }
+
+    private void compareNumber(RandomNumber randomNum, InputNumber inputNum) {
+        List<Integer> randomNumbers = randomNum.getNumbers().getRandomNumbers();
+        List<Integer> inputNumbers = inputNum.getNumbers().getInputNumbers();
+
+        for (int i = 0; i < randomNumbers.size(); i++) {
+            if (randomNumbers.get(i).equals(inputNumbers.get(i))) {
+                strike++;
+                continue;
+            }
+            if (randomNumbers.contains(inputNumbers.get(i))) {
+                ball++;
+            }
+        }
+    }
```
### 객체 지향 생활 체조 원칙
- 한 메서드에 오직 한 단계의 들여 쓰기만 한다.
- else 예약어를 쓰지 않는다.
- 모든 원시 값과 문자열을 포장한다.
- 한 줄에 점을 하나만 찍는다.
- 줄여 쓰지 않는다(축약 금지).
- 모든 엔티티를 작게 유지한다.
- 3개 이상의 인스턴스 변수를 가진 클래스를 쓰지 않는다.
- 일급 컬렉션을 쓴다.
- getter/setter/프로퍼티를 쓰지 않는다.

DongGeon0908 commented on src/main/java/baseball/run/RunGame.java:
```diff
@@ -0,0 +1,32 @@
+    private RandomNumber randomNum;
+    private final Output output = new Output();
+
+    public RunGame() {
+        generateRandomNum();
+    }
+
+    private void generateRandomNum(){
+        randomNum = new RandomNumber();
+    }
```
```suggestion
    public RunGame() {
        this. randomNum = new RandomNumber();
    }
```

DongGeon0908 replied:
메서드를 나누는건 좋은데, 너무 극한으로 나누는 것 같네요~

DongGeon0908 commented on src/main/java/baseball/run/RunGame.java:
```diff
@@ -0,0 +1,32 @@
+    public void run() {
+        CompareNumber compareNumber = new CompareNumber();
+
+        do {
+            compareNumber.setValue(randomNum);
+            output.output(outputResult(compareNumber));
+        } while (!compareNumber.isCheckStrike());
```
do while을 지양!

DongGeon0908 commented on src/main/java/baseball/run/RunGame.java:
```diff
@@ -0,0 +1,32 @@
+        output.output("BASEBALL GAME END!");
+    }
+
+    private String outputResult(CompareNumber compareNumber) {
+        return compareNumber.isCheckStrike() ? "3 Strike !" : "Strike : " + compareNumber.getStrike() + ", Ball : " + compareNumber.getBall();
```
삼항 연산자 금지!

추가적으로 해당 로직은 다른 곳에서 만들어주는게 어떨까요?

DongGeon0908 reviewed (commented):
고생하셨습니다!!

올려드린 객체지향 생활체조 원칙을 한번 읽어보시길!

InJun2 committed:
피드백 내용 적용
- 메소드명 변경
- RandomNumbers 클래스 제거
- while 문 제거 -> 재귀함수로 변경 ( 좋지는 않아 보임 <- 불필요 메모리를 더 잡아먹을 것 같음 )
- strike 결과 문자열 반환 메소드 CompareNumber 클래스로 변경
- InputNumbers 생성자 호출시 InputNumbers 에서 InputNumber 호출하여 필드 저장
- 자주 사용하는 변수 메소드 지역 변수 -> 클래스 전역 변수로 변경

InJun2 committed:
Merge pull request #9 from InJun2/injun/work

피드백 내용 적용

DongGeon0908 commented on src/main/java/baseball/run/CompareNumber.java:
```diff
@@ -0,0 +1,41 @@
+        return strike == THREE;
+    }
+
+    public String outputResult() {
+        if (isCheckStrike()) {
+            return "3 Strike !";
+        }
+
+        return "Strike : " + strike + ", Ball : " + ball;
+    }
```
strike와 ball을 반환하고, String으로 조립하는건 output쪽에서 하는게 좋을 듯 하네요!

DongGeon0908 commented on src/main/java/baseball/random/RandomNumber.java:
```diff
@@ -0,0 +1,25 @@
+public class RandomNumber {
+    private final List<Integer> randomNumbers;
+
+    public RandomNumber() {
+        Set<Integer> deduplicationNumbers = new LinkedHashSet<>();
+
+        while (deduplicationNumbers.size() < 3) {
+            deduplicationNumbers.add((int) (Math.random() * 9 + 1));
+        }
+
+        randomNumbers = new ArrayList<>(deduplicationNumbers);
+    }
```
생성자에서 비즈니스로직이 돌아가네요!
메서드로 분리하시죠

DongGeon0908 commented on src/main/java/baseball/input/InputNumbers.java:
```diff
@@ -0,0 +1,18 @@
+public class InputNumbers {
+    private final List<Integer> inputNumbers;
+
+    public InputNumbers() {
+        inputNumbers = new ArrayList<>(new InputNumber().deduplicationNum());
+    }
```
리스트 생성할때 초기화 시키는 건 오케이!
그런데 InputNumber를 여기서 생성하는게 맞을까요?
값만 생성자로 받아서 초기화해도 될 것 같은데

DongGeon0908 reviewed (commented):
생성자가 너무 많은 일을 하고 있네요!!
해당 부분만 수정하시고, 다음 단계로 넘어가시죠!!

다음 단계는 https://github.com/selab-hs/selab-todo-list 입니다!

InJun2 committed:
피드백 내용 적용
- InputNumbers 생성자 변경
- Output 클래스에서 결과 호출 메서드 위치
- RandomNumber 생성자 로직 메서드로 변경

InJun2 committed:
Merge pull request #10 from InJun2/injun/work

피드백 내용 적용
